### ISSUE TITLE
Electron example fails with "node-pre-gyp info This Node instance does not support builds for N-API version 4"

### ISSUE THREAD

**Issue by ksabr** [tensorflow/tfjs#2558]
Hi there,

I'm running into the issue described in #2370 when trying to run the `electron` example in `tfjs-examples`, but the recipes given there do not appear to help. Some more info below:

#### TensorFlow.js version

1.3.2

#### Browser version

74.0.3729.131 

#### Node version

12.9.1 (also tried 10.13.0, 10.16.3)

#### Platform

CentOS Linux release 7.6.1810 (Core) 

#### Describe the problem or feature request

Running the electron example at https://github.com/tensorflow/tfjs-examples/tree/master/electron results in:

```
node-pre-gyp info This Node instance does not support builds for N-API version 4
node-pre-gyp info This Node instance does not support builds for N-API version 4
App threw an error during load
Error: Cannot find module '/home/research/tfjs-examples/electron/node_modules/@tensorflow/tfjs-node/lib/napi-v3/tfjs_binding.node'
    at Module._resolveFilename (internal/modules/cjs/loader.js:584:15)
    at Function.Module._resolveFilename (/home/research/tfjs-examples/electron/node_modules/electron/dist/resources/electron.asar/common/reset-search-paths.js:43:12)
    at Function.Module._load (internal/modules/cjs/loader.js:510:25)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/home/research/tfjs-examples/electron/node_modules/@tensorflow/tfjs-node/dist/index.js:44:16)
    at Object.<anonymous> (/home/research/tfjs-examples/electron/node_modules/@tensorflow/tfjs-node/dist/index.js:66:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.require.extensions.(anonymous function) [as .js] (/home/research/tfjs-examples/electron/node_modules/electron-compile/lib/require-hook.js:77:14)
    at Module.load (internal/modules/cjs/loader.js:602:32)
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot find module '/home/research/tfjs-examples/electron/node_modules/@tensorflow/tfjs-node/lib/napi-v3/tfjs_binding.node'
    at Module._resolveFilename (internal/modules/cjs/loader.js:584:15)
    at Function.Module._resolveFilename (/home/research/tfjs-examples/electron/node_modules/electron/dist/resources/electron.asar/common/reset-search-paths.js:43:12)
    at Function.Module._load (internal/modules/cjs/loader.js:510:25)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/home/research/tfjs-examples/electron/node_modules/@tensorflow/tfjs-node/dist/index.js:44:16)
    at Object.<anonymous> (/home/research/tfjs-examples/electron/node_modules/@tensorflow/tfjs-node/dist/index.js:66:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.require.extensions.(anonymous function) [as .js] (/home/research/tfjs-examples/electron/node_modules/electron-compile/lib/require-hook.js:77:14)
    at Module.load (internal/modules/cjs/loader.js:602:32)
```
----

`npm rebuild @tensorflow/tfjs-node --build-from-source` seems to run fine but doesn't fix the problem. The `mnist-node` example runs without issues. Please let me know if you have any idea what could be going wrong.

**Comment by kangyizhang** [comment]
hi @ksabr , can you take a look of what is in the path `/home/research/tfjs-examples/electron/node_modules/@tensorflow/tfjs-node/lib`?

**Comment by ksabr** [comment]
Hi @kangyizhang ,

There's a directory `napi-v4` in there so clearly something is confusing `node-pre-gyp`. Looking into it a bit deeper, I discovered that modifying the `module.exports.get_napi_version` function in `node_modules/node-pre-gyp/lib/util/napi.js` (see https://github.com/mapbox/node-pre-gyp/blob/ef634f92f562cde064fbf831896fef045a47f9a1/lib/util/napi.js#L29) manually to return the value '4' makes the code run. As it is, the function returns '3' for some mysterious reason. This is despite this piece of code returning 4 as I'd expect:

```
var version = process.versions.napi;
console.log(version);
```
Any ideas why `node-pre-gyp` is behaving this way? Many thanks!

**Comment by kangyizhang** [comment]
node version 12.9.1 should be napi v5. can you try node version 12.13.0 and @tensorflow/tfjs-node version 1.4.0?

**Comment by ksabr** [comment]
Hi @kangyizhang ,

With node 12.13.0 and `    "@tensorflow/tfjs-node": "^1.4.0",` in `package.json` I get:

```
yarn run v1.21.1
$ electron .
node-pre-gyp info This Node instance does not support builds for N-API version 4
node-pre-gyp info This Node instance does not support builds for N-API version 5
node-pre-gyp info This Node instance does not support builds for N-API version 4
node-pre-gyp info This Node instance does not support builds for N-API version 5
App threw an error during load
```
For what it's worth, I now have to change `module.exports.get_napi_version` to return 5 to get it to run (i.e. return 4 no longer works).

The below prints 5 as expected:
```
var version = process.versions.napi;
console.log(version);
```

**Comment by ksabr** [comment]
Hi again @kangyizhang ,

I noticed that printing `process.versions` from `node_modules/node-pre-gyp/lib/util/napi.js` shows  `node: '10.11.0'` (whereas `node --version` gives `v12.13.0`). 10.11.0 is indeed NAPI v3, so that could explain the error. While trying to track down where 10.11.0 comes from, I noticed that `electron-prebuilt-compile@^4.0.0` (see https://github.com/tensorflow/tfjs-examples/blob/master/electron/yarn.lock#L2402) and `electron 4.0.0` itself require node 10.11.0: https://electronjs.org/releases/stable?version=4&page=6#4.0.0

It doesn't look like `electron-prebuilt-compile` has been maintained for a year now. I'm wondering whether this explains my issue, but if it does, I don't see how the code would work for anyone else either :smiley:

**Comment by kangyizhang** [comment]
@ksabr I can reproduce what you said. This example is suppose to show users how to use tfjs-node with electron framework, so it's possible that it does not work on all platforms. 

On windows, you will need to install a certain version of Node.js which has NAPI v3 and >= v8.12.0 (https://nodejs.org/api/n-api.html#n_api_n_api_version_matrix). You can try with node.js v8.12.0 and it works on my machine.

**Comment by ksabr** [comment]
Hi @kangyizhang ,

Thanks a lot - I have a working solution (installing electron 10.11.0, which is the one shipped with electron 4.0.0 used in the example), so I think we can close this. Hopefully other people running with the same problem on their platforms will be able to find a solution in this thread.

Many thanks again!
